packages/player/src/components/player.tsx

```diff
@@ -0,0 +1,461 @@
+  if (!state.context.video) return 0
+  const {currentTime, duration} = state.context.video
+  const {seekingTime} = state.context
+  const time = seekingTime || currentTime
+  const percent = time / duration
+  return percent >= 1 ? 1 : percent
+}
+
+const selectCurrentTime = (state: {context: VideoStateContext}) =>
+  state.context.video?.currentTime || 0
+
+const selectDuration = (state: {context: VideoStateContext}) =>
+  state.context.video?.duration || 0
+
+const SeekBar: React.FC<any> = (props) => {
+  const sliderRef = React.useRef(null)
+  const {videoService} = React.useContext(VideoContext)
+  const formattedTime = useSelector(videoService, selectFormattedTime)
+  const percent = useSelector(videoService, selectPercent)
+  const duration = useSelector(videoService, selectDuration)
+  const currentTime = useSelector(videoService, selectCurrentTime)
+
+  function calculateDistance(event: Event | SyntheticEvent) {
+    const node = sliderRef.current
+    const position = getPointerPosition(node, event)
+    return position.x
+  }
+
+  function getNewTime(event: Event | SyntheticEvent) {
+    const distance = calculateDistance(event)
+    const newTime = distance * duration
+
+    // Don't let video end while scrubbing.
+    return newTime === duration ? newTime - 0.1 : newTime
+  }
+
+  function handleMouseDown() {}
+
+  function handleMouseUp(event: Event | SyntheticEvent) {
+    const newTime = getNewTime(event)
+    // Set new time (tell video to seek to new time)
+    // actions.seek(newTime)
+    // actions.handleEndSeeking(newTime)
+    videoService.send({type: 'SEEKING', seekingTime: newTime})
+  }
+
+  function handleMouseMove(event: Event | SyntheticEvent) {
+    const newTime = getNewTime(event)
+    videoService.send({type: 'SEEKING', seekingTime: newTime})
+  }
+
+  function stepForward() {
+    let newTime = currentTime + 10
+    if (newTime > duration) {
+      newTime = duration
+    }
+    videoService.send({type: 'SEEKING', seekingTime: newTime})
+  }
+
+  function stepBack() {
+    let newTime = currentTime - 10
+    if (newTime < 0) {
+      newTime = 0
+    }
+    videoService.send({type: 'SEEKING', seekingTime: newTime})
+  }
+
+  return (
+    <Slider
+      ref={sliderRef}
+      label="video progress bar"
+      className={cx('cueplayer-react-progress-holder', props.className)}
+      valuenow={(percent * 100).toFixed(2)}
+      valuetext={formattedTime}
+      onMouseDown={handleMouseDown}
+      onMouseMove={handleMouseMove}
+      onMouseUp={handleMouseUp}
+      getPercent={() => percent}
+      stepForward={stepForward}
+      stepBack={stepBack}
+    >
+      {/*<LoadProgressBar*/}
+      {/*  buffered={buffered}*/}
+      {/*  currentTime={time}*/}
+      {/*  duration={duration}*/}
+      {/*/>*/}
+      {/*<MouseTimeDisplay duration={duration} mouseTime={mouseTime} />*/}
+      {/*<PlayProgressBar currentTime={time} duration={duration} />*/}
```

Comment:
TODO: port these components from cueplayer as function components and typescript